content/blog/2022-03-31-training-and-saving-models-on-a-dedicated-runner.md

```diff
@@ -0,0 +1,278 @@
+          cml runner \
+              --cloud=aws \
+              --cloud-region=eu-west \
+              --cloud-type=t2.micro \
+              --single
+  train-model:
+    needs: deploy-runner
+    runs-on: [self-hosted]
+    timeout-minutes: 120 # 2h
+    container:
+      image: docker://iterativeai/cml:0-dvc2-base1
+    steps:
+      - uses: actions/checkout@v2
+      - name: Train model
+        env:
+          REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+        run: |
+          cml ci
+          pip install -r requirements.txt
+          python get_data.py
+          python train.py
+```
+
+<admon type="warn">
+In this example we are using a <code>t2.micro</code> AWS EC2 instance. At the time of writing this is included in the AWS free tier. Make sure that you qualify for this free usage to prevent unexpected spending. When you specify a bulkier <code>cloud-type</code>, your expenses will rise.
+</admon>
+
+The workflow we defined first [provisions a runner](https://cml.dev/doc/ref/runner) on AWS, and then uses that
+runner to train the model. After completing the training job, CML automatically
+terminates the runner to prevent you from incurring further costs. Once the
+runner is terminated, however, the model is lost along with it. Let's see how we
+can save our model in the next step!
+
+# Export the model to our Git repository
+
+CML allows us to export the model from our runner to our Git repository. Let's
+extend the training stage of our workflow by pushing `random_forest.joblib` to a
+new experiment branch and creating a merge request.
+
+[`cml pr`](https://cml.dev/doc/ref/pr) is the command that specifies which files should be included in the
+pull request. The commands after that are used to generate a report in the
+pull request that displays the confusion matrix and calculated metrics.
+
+```yaml
+train-model:
+  needs: deploy-runner
+  runs-on: [self-hosted]
+  timeout-minutes: 120 # 2h
+  container:
+    image: docker://iterativeai/cml:0-dvc2-base1
+  steps:
+    - uses: actions/checkout@v2
+    - name: Train model
+      env:
+        REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+      run: |
+        cml ci
+        pip install -r requirements.txt
+        python get_data.py
+        python train.py
+
+        # Create pull request
+        cml pr model/random_forest.joblib
+
+        # Create CML report
+        cat model/metrics.txt > report.md
+        cml publish model/confusion_matrix.png --md >> report.md
+        cml send-comment --pr --update report.md
+```
+
+Et voilà! We are now running a daily model training on an AWS EC2 instance and
+saving the resulting model to our GitHub repository.
+
+There is still some room for improvement, though. This approach works well when
+our resulting model is small, but we wouldn't want to store large models in our
+Git repository. In a follow-up post we will describe how we can use
+[DVC](https://dvc.org/), another Iterative product, for storage when we're
```

Comment:
let's change "another Iterative product" to be more specific -> "another Iterative open-source tool"